networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/DiscoveryNetwork.java

```diff
@@ -156,6 +158,9 @@ public void setPreGenesisForkInfo() {
   }
 
   public void setForkInfo(final ForkInfo currentForkInfo, final Optional<Fork> nextForkInfo) {
+    // Remove the old fork info peer predicate
+    enrFieldPeerPredicateId.ifPresent(connectionManager::removePeerPredicate);
```

Comment:
We've removed the predicate and left a period where we aren't filtering peers.  We probably need this to be a replacePredicate atomic operation.

But it's probably simpler to just register one predicate with the connection manager that looks at a `volatile` field which stores the current `EnrForkId` and filters on that.  Then when the fork ID is updated, it updates that field and the existing predicate begins using the new value.

Reply:
Done.